### PR TITLE
WIP: various Github CI improvements for quilc

### PR DESCRIPTION
**Pull request by mhdavid-hrl**
various Github CI improvements for quilc

Makefile
  - add QVM to systems included by target dump-version-info target

.github/workflows/test.yml:
  - unify all jobs to run the same way except for make target
  - jobs now build with latest magicl and qvm cloned from official repo

### CODE REVIEW

**Comment by mhdavid-hrl** [comment]
Spun off from PR "add performance tweaks, primarily addresser-state caching, and some s… #757"

File: Makefile [hunk 1]
```diff
@@ -118,16 +122,25 @@ uninstall:
 test: test-cl-quil test-quilt test-quilec test-quilc
 
 test-cl-quil:
+# For test-cl-quil pull QVM and Magicl from appropriate repo/refspec.
+	$(GIT-PULL-QVM)
+	$(GIT-PULL-MAGICL)
```
Comment (kilimanjaro) [inline Makefile:127]: It's probably worth having these commands part of a different target (perhaps `get-latest-quilc-dependencies` or something). That way we can easily run tests with or without pulling the latest dependencies; without will just use whatever is in the quicklisp dist. You'd need to edit your workflow yaml to explicitly do `make get-latest-quilc-dependencies` for the relevant CI tasks.

File: Makefile [hunk 2]
```diff
@@ -114,9 +118,18 @@ uninstall:
...
-.PHONY: test test-cl-quil test-quilc test-quilt test-quilec
+.PHONY: test test-cl-quil test-quilc test-quilt test-quilec get_latest_quilc_deps
 test: test-cl-quil test-quilt test-quilec test-quilc
 
+# When testing, you may wish to pull QVM and Magicl from appropriate
+# repo/refspec prior to test. NOTE: this has side effects on
+# RIGETTI_LISP_LIBRARY_HOME directory, namely cloning deps
+# directories. Generally, this is meant to be run in a VM, typically
+# Docker-supplied, with an initially empty build directory.
+get_latest_quilc_deps:
```
Comment (kilimanjaro) [inline Makefile:129]:
```suggestion
get-latest-quilc-deps:
```
(just for consistency with the other targets)

**Comment by mhdavid-hrl** [comment]
The last (force) push really simplifies this a lot. Now, basically, for all the tests we just clone qvm and magicl into the quicklisp local project directory.

**Comment by kilimanjaro** [comment]
> The last (force) push really simplifies this a lot. Now, basically, for all the tests we just clone qvm and magicl into the quicklisp local project directory.

This is looking good.

**Comment by mhdavid-hrl** [comment]
This is quite simple now. We don't really need all the history of how this evolved. And I've been invited to make non-WIP version of this, so I'm going to create a fresh, non-WIP PR and close this. Closing now.

**Pull request closed without merging**